src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -8,7 +8,57 @@ public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        try {
+            loginNullCheck(user.getLogin());
+            passwordNullCheck(user.getPassword());
+            dublicateCheck(user);
+            loginLengthCheck(user.getLogin());
+            passwordLegthCheck(user.getPassword());
+            ageCheck(user.getAge());
+        } catch (RegistrationException e) {
+            throw e;
+        } catch (RuntimeException e) {
+            throw new RuntimeException(e);
```

Comment:
you don't need try-catch, declaring
![image](https://github.com/mate-academy/jv-registration-validation/assets/49767049/34265e59-0bfc-4f71-8214-4b6c8d6f8223)